### PR TITLE
DATAES-598:  Add Elasticsearch authentication a code example to readme

### PR DESCRIPTION
**Pull request by quxinyong**
…assword)

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@quxinyong Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=289)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=289) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@quxinyong Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-elasticsearch&pullRequestId=289)!

File: README.adoc [hunk 1]
```diff
@@ -243,6 +243,25 @@ public class Config {
   }
 }
 
+If the Elasticesearch Cluster has a BASIC_AUTH (user and password), You can do it like this.
```
Comment (sothawo) [inline README.adoc]:
```suggestion
//  If necessary, a BASIC_AUTH (user and password) header can be set with
```

File: README.adoc [hunk 2]
```diff
@@ -243,6 +243,25 @@ public class Config {
...
+@Configuration
+public class Config2 {
+	private static final String BASIC_AUTH = "Basic "
+			+ Base64.getEncoder().encodeToString("elastic:xihuaxian".getBytes(StandardCharsets.UTF_8));
+
+	@Bean
+	public ReactiveElasticsearchClient reactiveElasticsearchClient() {
+
+		HttpHeaders defaultHeaders = new HttpHeaders();
+		defaultHeaders.add(HttpHeaders.AUTHORIZATION, BASIC_AUTH);
```
Comment (sothawo) [inline README.adoc]:
```suggestion
		defaultHeaders.setBasicAuth(USER_NAME, USER_PASS);
```
better let the `HttpHeaders` class do the encoding

File: README.adoc [hunk 3]
```diff
@@ -243,6 +243,25 @@ public class Config {
   }
 }
 
+If the Elasticesearch Cluster has a BASIC_AUTH (user and password), You can do it like this.
+
+@Configuration
+public class Config2 {
+	private static final String BASIC_AUTH = "Basic "
```
Comment (sothawo) [inline README.adoc]: dont' t need that variable, see below

File: README.adoc [hunk 4]
```diff
@@ -243,6 +243,25 @@ public class Config {
   }
 }
 
+If the Elasticesearch Cluster has a BASIC_AUTH (user and password), You can do it like this.
+
+@Configuration
+public class Config2 {
```
Comment (sothawo) [inline README.adoc]:
```suggestion
public class ConfigWithAuthentication {
```

**Comment by quxinyong** [comment]
@sothawo ，I'm done with it.

**Comment by sothawo** [comment]
This PR has been merged. Thank you for your contribution!

**Pull request merged**